src/components/pages/Gallery/Gallery.tsx

```diff
@@ -0,0 +1,23 @@
+import PageTitle from "../../common/PageTitle/PageTitle";
+import GallerySet from "../../features/GallerySet/GallerySet";
+import ScrollTopButton from "../../features/ScrollTopButton/ScrollTopButton";
+import galleryData from "./galleryData";
+import useScroll from "../../../hooks/useScroll";
+import useTitle from "../../../hooks/useTitle";
+
+const Gallery: React.FC = () => {
+  useScroll();
```

Comment:
Z tego, co widzę, to każda zmiana podstrony ma powodować scrollTop. Można więc ustawić `useEffect` w App, który będzie wychwytywał zmianę lokacji i wtedy uruchamiał scrollTop. W takim przypadku miałabyś tę logikę tylko w App, a nie musiałabyś jej powielać w każdym komponencie podstrony.